src/Http/Controllers/ExportController.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace HeadlessLaravel\Formations\Http\Controllers;
+
+use HeadlessLaravel\Formations\Field;
+use HeadlessLaravel\Formations\Formation;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Route;
+use Maatwebsite\Excel\Facades\Excel;
+
+class ExportController
+{
+    public function create(Request $request)
+    {
+        /** @var Formation $formation */
+        $formation = app(Route::current()->parameter('formation'));
+
+        $exportable = $formation->exportable();
+        $requestFields = $request->get('columns');
+
+        if (!empty($requestFields)) {
+            $requestFields = explode(',', $requestFields);
+            $exportable->fields = collect($formation->export())->filter(function (Field $field) use ($requestFields) {
+                return in_array($field->key, $requestFields);
+            })->toArray();
```

Comment:
We only apply columns if columns are passed. so I think 

> Provides feedback to the frontend instead of allowing empty downloads if 0 valid columns

is not needed. And if an invalid column name is passed then we throw 422 error.